src/jit/codegenxarch.cpp

```diff
@@ -7549,6 +7556,232 @@ void CodeGen::genSetRegToCond(regNumber dstReg, GenTreePtr tree)
     }
 }
 
+
+void CodeGen::genOverflowCheck_I8toI4(regNumber hiReg, regNumber loReg)
+{
+
+    // conv.ovf.i8.i4
+    // Generate the following sequence
+    // test loDWord, loDWord   // set flags
+    // jl neg
+    // pos: test hiDWord, hiDWord   // set flags
+    // jne ovf
+    // jmp done
+    // neg: cmp hiDWord, 0xFFFFFFFF
+    // jne ovf
+    // done:
+
+    emitJumpKind jmpNotEqual = genJumpKindForOper(GT_NE, CK_SIGNED);
+    emitJumpKind jmpLTS = genJumpKindForOper(GT_LT, CK_SIGNED);
+
+    instGen_Compare_Reg_To_Zero(EA_4BYTE, loReg);
+
+    BasicBlock * neg;
+    BasicBlock * done;
+
+    neg = genCreateTempLabel();
+    done = genCreateTempLabel();
+
+    // Is the loDWord positive or negative
+    inst_JMP(jmpLTS, neg);
+
+    // If loDWord is positive, hiDWord should be 0 (sign extended loDWord)
+    instGen_Compare_Reg_To_Zero(EA_4BYTE, hiReg);
+
+    genJumpToThrowHlpBlk(jmpNotEqual, SCK_OVERFLOW);
+    inst_JMP(EJ_jmp, done);
+
+    // If loDWord is negative, hiDWord should be -1 (sign extended loDWord)
+
+    genDefineTempLabel(neg);
+
+    inst_RV_IV(INS_cmp, hiReg, 0xFFFFFFFFL, EA_4BYTE);
+
+    genJumpToThrowHlpBlk(jmpNotEqual, SCK_OVERFLOW);
+
+    // Done
+
+    genDefineTempLabel(done);
+}
+
+void CodeGen::genOverflowCheck_I8toU4(regNumber hiReg, regNumber loReg)
+{
+    // Just check that the upper DWord is 0
+
+    instGen_Compare_Reg_To_Zero(EA_4BYTE, hiReg); // set flags
+
+    emitJumpKind jmpNotEqual = genJumpKindForOper(GT_NE, CK_SIGNED);
+    genJumpToThrowHlpBlk(jmpNotEqual, SCK_OVERFLOW);
+}
+
+void CodeGen::genOverflowCheck_U8toI4(regNumber hiReg, regNumber loReg)
+{
+    // ; if low_word < 0 raise overflow
+    // 02D02DE3  mov         eax, dword ptr ds : [0169C8E8h]
+    // 02D02DE8  test        eax, eax
+    // 02D02DEA  jl          02D02DF9
+    // ; if high_word != 0 raise overflow
+    // 02D02DEC  cmp         dword ptr ds : [169C8ECh], 0
+    // 02D02DF3  jne         02D02DF9
+    // 02D02DF5  pop         ebp
+    // 02D02DF6  ret         8
+    // ; overflow exception
+    // 02D02DF9  call        73BECA1D 
```

Comment:
Oops, sorry about this. It looks like I run my test in 64 bit mode instead of 32 bit and failed to notice. It seemed too good to be true but I guess I was too tired to see. Same about the I8 to I4.